Incremental improvements - Pull request to merge to master

jscarsdale1 opened the pull request:
Update to Angular 13 and Node 16.

tompatman commented on README.md:
```diff
@@ -12,30 +19,33 @@ of sudo privlages. Make sure that you can ssh as service to the target without a
-- From the phpstorm `npm` window, select `build-client-dev` to produce a dist.tgz file.
-- In the phpstorm `npm` window, click `copy-dist-to-output` to copy the `dist.tgz` file to `../../../output/simpleui/base_app` for consumption by the deploy script.
+- From the JetBrains IDE's `npm` window, select `build-client-dev` to produce a dist.tgz file.
+- In the JetBrains IDE's `npm` window, click `copy-dist-to-output` to copy the `dist.tgz` file to `../../../output/simpleui/base_app` for consumption by the deploy script.
```
My only comment on this doc is the use (requirement?)  of Jetbrains. That works for our purposes but if this is staying open source these directions should be using npm commands in the command line. Just a note for the future.

tompatman commented on base_app/post-build-step.bash:
```diff
@@ -48,25 +48,35 @@ cat dist/simpleui/index.html | \
-#ln -s /var/www/${PROJ}/sample_ui.properties.txt /var/www/${PROJ}/ui.properties
-ln -s /var/www/${PROJ}/mock-data/ui.bms.mock.properties /var/www/${PROJ}/ui.properties
+declare -A true_words_array=( [yes]=true [true]=true [1]=true [t]=true )
+if [[ -v true_words_array[${SUI_BUILD_WILL_UPDATE_LOCAL_VAR_WWW,,}] ]]; then
```
How is this getting set? Is this an environment variable and does that mean the ansible update scripts need to incorporate it?

jscarsdale1 replied:
The SUI_BUILD_WILL_UPDATE_LOCAL_VAR_WWW env var test was added to avoid developers inadvertently updating their build machine's `/var/www/simpleui folder` when doing `npm run build_prod` (or whatever) from console or IDE. This section would only be run when normally shouldn't be run at all.

Since I'm likely the only consumer, and not actively using the feature, I'm going to remove the entire block from the script.

tompatman commented on base_app/src/app/app.component.ts:
```diff
@@ -621,30 +602,6 @@ export class AppComponent implements OnInit, AfterViewInit /*, OnChanges */ {
             });
     }
 
-    displayRefreshPausedMessage() {
```
Is this being deleted or moved elsewhere? Don't we use this pause feature?

tompatman commented on base_app/src/tools/logger.ts:
```diff
@@ -139,12 +136,12 @@ export class ClientLogger {
     CRITICAL,
-    ERROR,
-    WARNING,
+//    ERROR,
```
If these are no longer needed remove instead of comment out. But wouldn't an error level still be needed?

jscarsdale1 replied:
I think we're not currently using ERROR in the typescript logger.  It's commented because it seems likely to be used at some point, but don't want it to show "unused" in the IDE.

tompatman commented on web/html/ParamsApp/Parameters.html:
```diff
@@ -13,7 +13,7 @@
 
 <BODY>
-    <h1><img src="img/vionx-energy-logo-2.png" alt="Vionx logo" height="43" width="166">Cross-Module Parameter Manager</h1>
+    <h1><img src="img/largo-inc-logo-2.png" alt="Largo Inc logo" height="43" width="166">Parameter Manager</h1>
```
Also, for future work, it occurs to me that we need to get the company specific stuff out of the open source project.

ntosta commented on .idea/misc.xml:
```diff
@@ -3,4 +3,14 @@
   <component name="JavaScriptSettings">
     <option name="languageLevel" value="ES6" />
   </component>
+  <component name="UnattendedHostPersistenceState">
+    <option name="openedFilesInfos">
+      <list>
+        <OpenedFileInfo>
+          <option name="caretOffset" value="129" />
+          <option name="fileUrl" value="file://$PROJECT_DIR$/base_app/src/index.html" />
+        </OpenedFileInfo>
+      </list>
+    </option>
+  </component>
```
Is there a specific reason why the .idea folder has been committed? Might be worth just deleting the .idea folder entirely and adding it to the gitignore

jscarsdale1 replied:
I'd like to keep the  dictionaries folder (or an alternative dictionary) so that `simpleui` and similar aren't highlighted as a misspellings by the IDE.

Updated `.gitignore` with the other filenames that don't need to be checked in.

ntosta commented on addons/uimanager.html:
```diff
@@ -0,0 +1,83 @@
+<table>
+    <tr><th>Tab</th><th>Last Refresh</th><th>Memory</th><th>Action</th></tr>
+    <tr>
+        <td id="pureLink">Purification</td>
```
This file has a bunch of purification-specific stuff in it, is there a reason it has to be in this repo?

jscarsdale1 replied:
Deleted this file - no longer necessary since I verified that `minutesBeforeAutoPageReload` (new name) works to accomplish the same function.

ntosta commented on base_app/src/app/app-tab-overlay/app-tab-overlay.component.ts:
```diff
@@ -392,16 +400,21 @@ export class AppTabOverlayComponent implements AfterViewInit, OnInit {
+        const newTop: any = Math.round(ev.pageY - anchorCoords.top + moveLogger['dragStartInfo']['topOffsetToMouse']);
+        const newLeft: any = Math.round(ev.pageX - anchorCoords.left + moveLogger['dragStartInfo']['leftOffsetToMouse']);
+
+        let newHeight: any = e && e.children && e.children[1] && e.children[1].offsetHeight || e.offsetHeight;
```
you don't need all the ": any" type annotations, can just leave them out. In general 'any' should be used very sparingly since you're effectively telling the type system to ignore whatever you're doing.

jscarsdale1 replied:
Will change type to `number`.

Sadly we weren't (I wasn't) disciplined enough to use real types everywhere.
A lot of code was developed prototypically, starting simply from an object and adding fields as necessary, but  the follow up should have been to complete the working using exact types.

We need to scrub out the `any` types from the typescript, but probably won't enter a dedicated issue for it.
Prefer to do it as part of other refactoring / cleanup efforts.

ntosta commented on base_app/src/app/app-tab-overlay/implemented-overlays.service.ts:
```diff
@@ -51,7 +51,7 @@ export class ImplementedOverlaysService {
             .catch(this.handleError);
         */
-        return (null);
+        return null;
```
This function is annotated as returning a 'Observable<CSSPairList>' but here you return null. It looks like this function can't return anything other than null, in which case the function signature should be `getOverlays(): void` and you can omit the return statement.

jscarsdale1 replied:
This class is obsolete and is being removed .

ntosta commented on base_app/src/test/00.readme.unit-tests.md:
```diff
@@ -1,6 +1,9 @@
+## When to Execute
+ALWAYS EXECUTE THE UNIT TESTS AFTER THESE DEVELOPMENT ACTIVITIES:
+
+    a. After updatingany file in the src/public/nodejs folder
```
typo

ntosta commented:
I notice there's a ton of file changes such as `simpleui-server/test/script/sample-data-collection/collect-test-samples.bash`, do these actually need to be committed?

jscarsdale1 commented:
>I was mostly focused on moving those OUT of `base_app` (the client side).  Since those tests were used, we moved the `xml -> jscon` conversion into `simpleui-server`, which is why I moved them there.  Does it look like too much pollution?  Maybe should move into a test project?

ntosta commented:
> > I was mostly focused on moving those OUT of `base_app` (the client side).  Since those tests were used, we moved the `xml -> jscon` conversion into `simpleui-server`, which is why I moved them there.  Does it look like too much pollution?  Maybe should move into a test project?

Ah if they're actually used for testing that's fine, just wanted to make sure these are actually being used for testing and aren't just stuff that got committed by accident

jscarsdale1 commented on base_app/src/public/mock-data/purification-overlay.0.xml:
```diff
@@ -0,0 +1,310 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Yes - this is mock data used for testing without having a backend server available, and for verifying functionality.  Hope to integrate with a decent `e2e` test soon.

When that happens, many obsolete mock data files will be removed, but the ones that are still valid will remain.

jscarsdale1 commented on base_app/src/app/interfaces/popup-dialog-props.ts:
```diff
@@ -0,0 +1,14 @@
+/**
+ * Created by jscarsdale on 6/30/16.
+ */
```
removed.
